homework37/main.js

```diff
@@ -0,0 +1,116 @@
+import { isRequired } from "./modules/validation.js";
+import {showError, showSuccess, removeValidClass } from "./modules/formUI.js";
+
+const api_Endpoint = 'https://jsonplaceholder.typicode.com';
+const form = document.forms.getPostById;
+const postsInfo = document.querySelector(".posts__info");
+const commentSection = document.querySelector(".comments-section");
+const showComments = document.querySelector(".showComments");
+const hideComments = document.querySelector(".hideComments");
+
+const getIdNumInput = () => document.getElementById("idNum");
+
+const getIdNumValue = () => getIdNumInput().value.trim();
+
+const renderCartInfo = (data) => {
+    postsInfo.style.display = "block";
+
+    const titleOfPost = postsInfo.querySelector(".card-title");
+    const numOfPost = postsInfo.querySelector(".idVal");
+    const postText = postsInfo.querySelector(".card-text");
+
+    titleOfPost.innerHTML = `${data.title}`;
+    numOfPost.innerHTML = `${data.id}`;
+    postText.innerHTML = `${data.body}`;
+};
+
+const renderComments = (data) => {
+    data.forEach(element => {
+        commentSection.insertAdjacentHTML("beforeend", `
+        <h5 class="card-title">Комментар від ${element.name}</h5>
+        <h6 class="card-subtitle">${element.email}</h6>
+        <p class="card-text">${element.body}</p>`);
+    });
+};
+
+const displayError = (error) => {
+    alert(error.message);
+};
+
+const apiCall = (url) => fetch(url).then(res => res.json());
+
+const getPostById = (postId) => apiCall(`${api_Endpoint}/posts/${postId}`);
+
+const getCommentsByPostId = (postId) => apiCall(`${api_Endpoint}/posts/${postId}/comments`);
+
+const handlePost = (event) => {
+    event.preventDefault();
+
+    if (!checkOneToHundred(getIdNumInput(), getIdNumValue())) {
+        checkOneToHundred(getIdNumInput(), getIdNumValue());
+    } else {
```

Comment:
I don't see the reason to move to separate methods, because pretty simple

```
const getIdNumInput = () => document.getElementById("idNum");

// In this method you have duplicated call of getIdNumInput
const getIdNumValue = () => getIdNumInput().value.trim();
```

Much better
```
const element = document.getElementById("idNum");

checkOneToHundred(element, element.value.trim())
```